src/DisputeHandler.ts

```diff
@@ -55,164 +72,281 @@ class DisputeHandler {
+        )) as unknown as DisputeOutputState;
+
+        const participants =
+            (await this.stateChannelManagerContract.getStatemachineParticipants(
+                encodedModifiedState
+            )) as unknown as AddressLike[];
 
-        const newDispute = await this.stateChannelManagerContract.getDispute(
-            this.channelId
+        const disputeOutputStateSnapshot: StateSnapshotStruct = {
+            stateMachineStateHash: ethers.keccak256(
+                ethers.toUtf8Bytes(encodedModifiedState)
+            ),
+            participants,
+            forkCnt: forkCnt + 1,
+            latestExitChannelBlockHash,
+            latestJoinChannelBlockHash: onChainLatestJoinChannelBlockHash,
+            totalDeposits,
+            totalWithdrawals
+        };
+        const disputeOutputStateSnapshotHash = ethers.keccak256(
+            EvmUtils.encodeStateSnapshot(disputeOutputStateSnapshot)
         );
-        //TODO! check newDispute 0000 bytes
-        if (newDispute.channelId == ethers.ZeroHash) {
-            throw new Error(
-                "DisputeHandler - createDispute - no dispute created"
-            );
-        }
-        await this.rechallengeRecursive(newDispute);
+
+        const { dispute: defaultDispute, timeout: defaultTimeout } =
+            this.createDefaultDisputeStruct();
+
+        const disputeAuditingData = this.createDisputeAuditingData(
+            forkCnt,
+            transactionCnt,
+            disputeOutputStateSnapshot,
+            timestamp,
+            defaultDispute,
+            ethers.MaxInt256
+        );
+        const disputeAuditingDataHash = ethers.keccak256(
+            EvmUtils.encodeDisputeAuditingData(disputeAuditingData)
+        );
+        let timeoutDispute = timeout ?? defaultTimeout;
+        let dispute: DisputeStruct = {
+            channelId: this.channelId,
+            genesisStateSnapshotHash: genesisStateSnapshotHash,
+            latestStateSnapshotHash: latestStateSnapshotHash,
+            stateProof: stateProof,
+            fraudProofs: proofs,
+            onChainSlashes,
+            onChainLatestJoinChannelBlockHash,
+            outputStateSnapshotHash: disputeOutputStateSnapshotHash,
+            exitChannelBlocks,
+            disputeAuditingDataHash,
+            disputer: this.signerAddress,
+            disputeIndex: disputeIndex,
+            previousRecursiveDisputeIndex: ethers.MaxUint256,
+            timeout: timeoutDispute,
+            selfRemoval: false
+        };
+        return { dispute, disputeAuditingData };
     }
 
-    private async createNewDispute(
-        forkCnt: BigNumberish,
-        foldedParticipant: AddressLike,
-        foldedTransactionCnt: BigNumberish,
-        proofs: ProofStruct[]
+    public async createNewDispute(
+        forkCnt: number,
+        transactionCnt: number,
+        proofs: ProofStruct[],
+        timestamp: BigNumberish,
+        timeout?: TimeoutStruct
```

Comment:
This doesn't account for self removals and you have to account for joins in advance and assume they will succeed on-chain, otherwise the tx will revert anyway